File: 算法笔试/360_2/3.java
/*
 * @Descripttion: 
 * @Author: daxiong
 * @Date: 2019-08-31 17:46:16
 * @LastEditors: daxiong
 * @LastEditTime: 2019-08-31 17:46:16
 */;

public class Main {
 
    public static Set<Integer> set = new HashSet<>();
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
 
        int n = in.nextInt(), m = in.nextInt();
        int[] d = new int[m];
        for (int i = 0; i < m; i++) {
            d[i] = in.nextInt();
        }
 
        for (int i = 0; i < n; i++) {
            dfs(d, 0, i, n, i);
        }
        System.out.println(set.size());
    }
 
    private static void dfs(int[] d, int idx, int pos, int n, int fa) {
        if (idx >= d.length && pos >= 0 && pos < n) {
            set.add(pos);
            return;
        }
        if (d[idx] <= pos) {
            dfs(d, idx + 1, pos - d[idx], n, fa);
        }
 
        if (d[idx] <= n - pos - 1) {
            dfs(d, idx + 1, pos + d[idx], n, fa);
        }
    }
}

File: 算法笔试/招商银行/3.py
'''
@Descripttion: 
@Author: daxiong
@Date: 2019-09-15 20:30:18
@LastEditors: daxiong
@LastEditTime: 2019-09-15 22:42:51
'''
n = input()
ans = [1]* len(n)
if len(n) == 2:
    print('1 1')
else:
    for i in range(1, len(n)-1):
        if n[i] == 'R' and n[i-1] == 'R':
            ans[i+1] += ans[i-1]
            ans[i-1] = 0
    for j in range(len(n)-2, 0, -1):
        if n[j] == 'L' and n[j+1] == 'L':
            ans[j-1] += ans[j+1]
            ans[j+1] = 0
    print(' '.join(str(i) for i in ans))